src/native_driver.jl

```diff
@@ -40,12 +95,35 @@ function wait_for_pod_init(manager::K8sNativeManager, pod)
     throw(TimeoutException("timed out after waiting for worker $(pod.metadata.name) to init for $(manager.retry_seconds) seconds, with status\n $status", e))
 end
 
-function launch(manager::K8sNativeManager, params::Dict, launched::Array, c::Condition)
+function worker_pod_spec(manager::K8sClusterManager; kwargs...)
+    pod = worker_pod_spec(manager.ctx;
+                          driver_name=manager.driver_name,
+                          image=manager.image,
+                          cpu=manager.cpu,
+                          memory=manager.memory,
+                          kwargs...)
+
+    return pod
+end
+
+function Distributed.launch(manager::K8sClusterManager, params::Dict, launched::Array, c::Condition)
+    exename = params[:exename]
+    exeflags = params[:exeflags]
+
+    cmd = `$exename $exeflags --worker=$(cluster_cookie())`
+
     errors = Dict()
     # try not to overwhelm kubectl proxy; wait longer if more workers requested
-    sleeptime = 0.1 * sqrt(length(manager.pods))
-    asyncmap(collect(pairs(manager.pods))) do p
-        port, pod = p
+    sleeptime = 0.1 * sqrt(length(manager.ports))
+    asyncmap(manager.ports) do port
+        pod = @static if VERSION >= v"1.5"
+            worker_pod_spec(manager; port=port, cmd=cmd)
```

Comment:
Is there any downside to unconditionally using this form?

Reply:
Just not as nice to read. I was thinking of dropping support for older Julia versions here if it's kosher so this mainly serves as a reminder.